src/errors.rs

```diff
@@ -22,6 +22,16 @@ impl StdError for BodyError {
     fn description(&self) -> &str {
         &self.detail[..]
     }
+
+    fn cause(&self) -> Option<&StdError> {
+        use BodyErrorCause::*;
+
+        match self.cause {
```

Comment:
No, `.0` syntax only works for structs and tuple structs, but `cause` is an enum. What would the type of `self.cause.0` be? It depends on which variant you have.